homeassistant/components/light/mqtt.py

```diff
@@ -0,0 +1,237 @@
+"""
+homeassistant.components.light.mqtt
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Allows to configure a MQTT light.
+
+config for RGB Version with brightness:
+
+light:
+  platform: mqtt
+  name: "Office Light RGB"
+  state_topic: "office/rgb1/light/status"
+  command_topic: "office/rgb1/light/switch"
+  brightness_state_topic: "office/rgb1/brightness/status"
+  brightness_command_topic: "office/rgb1/brightness/set"
+  rgb_state_topic: "office/rgb1/rgb/status"
+  rgb_command_topic: "office/rgb1/rgb/set"
+  qos: 0
+  payload_on: "on"
+  payload_off: "off"
+
+config without RGB:
+
+light:
+  platform: mqtt
+  name: "Office Light"
+  state_topic: "office/rgb1/light/status"
+  command_topic: "office/rgb1/light/switch"
+  qos: 0
+  payload_on: "on"
+  payload_off: "off"
+
+"""
+import logging
+import homeassistant.components.mqtt as mqtt
+from homeassistant.components.light import (Light,
+                                            ATTR_BRIGHTNESS, ATTR_RGB_COLOR)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "MQTT Light"
+DEFAULT_QOS = 0
+DEFAULT_PAYLOAD_ON = "on"
+DEFAULT_PAYLOAD_OFF = "off"
+DEFAULT_RGB = [255, 255, 255]
+DEFAULT_RGB_PATTERN = "%d,%d,%d"
+DEFAULT_BRIGHTNESS = 120
+DEFAULT_OPTIMISTIC = False
+
+DEPENDENCIES = ['mqtt']
+
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Add MQTT Light. """
+
+    if config.get('command_topic') is None:
+        _LOGGER.error("Missing required variable: command_topic")
+        return False
+
+    if config.get('rgb_command_topic') is not None:
```

Comment:
Let's leave color temp out for now and get it in with the current functionality .